synapse/http/server.py

```diff
@@ -68,10 +68,21 @@
     "response_db_txn_count", labels=["method", "servlet", "tag"]
 )
 
+# seconds spent waiting for db txns, excluding scheduling time, when processing
+# this request
 response_db_txn_duration = metrics.register_distribution(
     "response_db_txn_duration", labels=["method", "servlet", "tag"]
 )
 
+# seconds spent waiting for a db connection, when processing this request
+#
+# it's a counter rather than a distribution, because the count would always
+# be the same as that of all the other distributions.
```

Comment:
It is a distribution here.

Honestly I think it makes things clearer to always include the count, as otherwise it loos a bit odd to do: `rate(metric_one:total)/rate(another_metric:count)`

Reply:
> It is a distribution here.

er, oops.

> Honestly I think it makes things clearer to always include the count, as otherwise it loos a bit odd to do: rate(metric_one:total)/rate(another_metric:count)

It seems really silly to me to maintain six identical copies of the same counter here. That's a lot of pointless objects, hash lookups, and integer increments. IMHO what we *ought* to be doing is `rate(synapse_http_server_response_db_sched_duration)/rate(synapse_http_server_response_count)`, which feels much more intuitive, but will take a bit of work to get there.

Reply:
> It seems really silly to me to maintain six identical copies of the same counter here. That's a lot of pointless objects, hash lookups, and integer increments.

I would be surprised if they're not completely dwarfed by transaction overhead.

> IMHO what we ought to be doing is `rate(synapse_http_server_response_db_sched_duration)/rate(synapse_http_server_response_count)`, which feels much more intuitive, but will take a bit of work to get there.

Possibly, but having things consistent seems more intuitive than having a couple that don't fit.